File: app/src/main/java/com/android/stanlyfang/chopstick/sample/di/component/AppComponent.java
package com.android.stanlyfang.chopstick.sample.di.component;

import com.android.stanlyfang.chopstick.sample.App;
import com.android.stanlyfang.chopstick.sample.di.module.ActivityModule;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.support.AndroidSupportInjectionModule;

@Component(modules = {AndroidInjectionModule.class,
        ActivityModule.class,
        AndroidSupportInjectionModule.class})
public interface AppComponent{

    void inject(App instance);
}

File: chopstick/src/main/java/com/android/stanlyfang/chopstick/di/component/AppComponent.kt
package com.android.stanlyfang.chopstick.di.component

import com.android.stanlyfang.chopstick.BaseApplication
import dagger.Component
import dagger.android.AndroidInjectionModule

@Component(modules = [(AndroidInjectionModule::class)])
interface AppComponent{
    fun inject(instance : BaseApplication)
}